clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -56,20 +56,19 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
         long startTime = time.milliseconds();
         long expiryTime = startTime + timeoutMs;
 
-        if (!isReady(client, node, startTime))
-            client.ready(node, startTime);
+        if (isReady(client, node, startTime) ||  client.ready(node, startTime))
```

Comment:
I don't think `immediate connections` is possible, but `isReady` could conceivably return `false` and then `true` and the second one (called from `ready`) could be missed without Jason's change.